File: packages/fixflow_core/lib/models/responses/job_status_history_response.dart
import 'package:fixflow_core/enums/job_status.dart';

class JobStatusHistoryResponse {
  final int id;
  final int bookingId;
  final JobStatus previousStatus;
  final JobStatus newStatus;
  final String? note;
  final int changedById;
  final String changedByFirstName;
  final String changedByLastName;
  final DateTime createdAt;

  JobStatusHistoryResponse({
    required this.id,
    required this.bookingId,
    required this.previousStatus,
    required this.newStatus,
    this.note,
    required this.changedById,
    required this.changedByFirstName,
    required this.changedByLastName,
    required this.createdAt,
  });

  String get changedByFullName => '$changedByFirstName $changedByLastName';

  factory JobStatusHistoryResponse.fromJson(Map<String, dynamic> json) {
    return JobStatusHistoryResponse(
      id: json['id'] as int,
      bookingId: json['bookingId'] as int,
      previousStatus: JobStatus.fromJson(json['previousStatus']),
      newStatus: JobStatus.fromJson(json['newStatus']),
      note: json['note'] as String?,
      changedById: json['changedById'] as int,
      changedByFirstName: json['changedByFirstName'] as String,
      changedByLastName: json['changedByLastName'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String).toLocal(),
    );
  }
}

File: FixFlow/FixFlow.Application/DTOs/Response/JobStatusHistoryResponse.cs
using FixFlow.Core.Enums;

namespace FixFlow.Application.DTOs.Response;

public class JobStatusHistoryResponse
{
    public int Id { get; set; }
    public int BookingId { get; set; }
    public JobStatus PreviousStatus { get; set; }
    public JobStatus NewStatus { get; set; }
    public string? Note { get; set; }
    public int ChangedById { get; set; }
    public string ChangedByFirstName { get; set; } = string.Empty;
    public string ChangedByLastName { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}
